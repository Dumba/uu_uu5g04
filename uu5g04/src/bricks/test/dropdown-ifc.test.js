/**
 * Copyright (C) 2021 Unicorn a.s.
 *
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public
 * License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License at
 * <https://gnu.org/licenses/> for more details.
 *
 * You may obtain additional information at <https://unicorn.com> or contact Unicorn a.s. at address: V Kapslovne 2767/2,
 * Praha 3, Czech Republic or at the email: info@unicorn.com.
 */

import UU5 from "uu5g04";
import "uu5g04-bricks";

const { mount, shallow, wait } = UU5.Test.Tools;

describe(`UU5.Bricks.Dropdown interface testing`, () => {
  it("isDropdown()", () => {
    const wrapper = shallow(
      <UU5.Bricks.Dropdown id={"uuID01"} label="Dropdown" size="l" colorSchema="blue" disableBackdrop>
        <UU5.Bricks.Dropdown.Item id={"uuID02"} label="Label" />
      </UU5.Bricks.Dropdown>
    );
    expect(wrapper.instance().isDropdown()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("open(setStateCallBack)", () => {
    const wrapper = shallow(
      <UU5.Bricks.Dropdown id={"uuID01"} label="Dropdown" size="l" colorSchema="blue" disableBackdrop>
        <UU5.Bricks.Dropdown.Item id={"uuID02"} label="Label" />
      </UU5.Bricks.Dropdown>
    );
    expect(wrapper).toMatchSnapshot();
    const mockFunc = jest.fn();
    expect(wrapper.instance().isOpen()).toBeFalsy();
    expect(wrapper.instance().state.open).toBeFalsy();
    const returnValue = wrapper.instance().open(mockFunc);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.open).toBeTruthy();
    expect(wrapper.instance().isOpen()).toBeTruthy();
    expect(mockFunc).toBeCalled();
    expect(mockFunc).toHaveBeenCalledTimes(1);
    expect(returnValue === wrapper.instance()).toBe(true);
  });

  it("close(setStateCallBack)", () => {
    const wrapper = shallow(
      <UU5.Bricks.Dropdown id={"uuID01"} label="Dropdown" size="l" colorSchema="blue" disableBackdrop>
        <UU5.Bricks.Dropdown.Item id={"uuID02"} label="Label" />
      </UU5.Bricks.Dropdown>
    );
    expect(wrapper).toMatchSnapshot();
    const mockFunc = jest.fn();
    expect(wrapper.instance().isOpen()).toBeFalsy();
    expect(wrapper.instance().state.open).toBeFalsy();
    const returnValue = wrapper.instance().open(mockFunc);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.open).toBeTruthy();
    expect(wrapper.instance().isOpen()).toBeTruthy();
    expect(mockFunc).toBeCalled();
    expect(mockFunc).toHaveBeenCalledTimes(1);
    expect(returnValue === wrapper.instance()).toBe(true);
    //Now we closed opened drop down
    const closeValue = wrapper.instance().close(mockFunc);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().isOpen()).toBeFalsy();
    expect(wrapper.instance().state.open).toBeFalsy();
    expect(mockFunc).toBeCalled();
    expect(mockFunc).toHaveBeenCalledTimes(2);
    expect(closeValue === wrapper.instance()).toBe(true);
  });

  it("toggle(setStateCallBack) first should open, second should close.", () => {
    const wrapper = shallow(
      <UU5.Bricks.Dropdown id={"uuID01"} label="Dropdown" size="l" colorSchema="blue" disableBackdrop>
        <UU5.Bricks.Dropdown.Item id={"uuID02"} label="Label" />
      </UU5.Bricks.Dropdown>
    );
    const mockFunc = jest.fn();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().isOpen()).toBeFalsy();
    const returnValue = wrapper.instance().toggle(mockFunc);
    wrapper.update();
    expect(wrapper.instance().isOpen()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
    const returnValue2 = wrapper.instance().toggle(mockFunc);
    wrapper.update();
    expect(wrapper.instance().isOpen()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("isOpen() should return false", () => {
    const wrapper = shallow(
      <UU5.Bricks.Dropdown id={"uuID01"} label="Dropdown" size="l" colorSchema="blue" disableBackdrop>
        <UU5.Bricks.Dropdown.Item id={"uuID02"} label="Label" />
      </UU5.Bricks.Dropdown>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().isOpen()).toBeFalsy();
  });

  it("isOpen() should return true", () => {
    const wrapper = shallow(
      <UU5.Bricks.Dropdown id={"uuID01"} label="Dropdown" size="l" colorSchema="blue" disableBackdrop>
        <UU5.Bricks.Dropdown.Item id={"uuID02"} label="Label" />
      </UU5.Bricks.Dropdown>
    );
    wrapper.instance().open();
    wrapper.update();
    expect(wrapper.instance().isOpen()).toBeTruthy();
  });
});
