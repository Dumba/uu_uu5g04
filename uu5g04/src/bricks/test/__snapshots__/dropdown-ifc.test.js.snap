// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Dropdown interface testing close(setStateCallBack) 1`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing close(setStateCallBack) 2`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-open uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
  <forwardRef(Popover)
    controlled={false}
    location="local"
  />
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing close(setStateCallBack) 3`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing isDropdown() 1`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing isOpen() should return false 1`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing open(setStateCallBack) 1`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing open(setStateCallBack) 2`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-open uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
  <forwardRef(Popover)
    controlled={false}
    location="local"
  />
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing toggle(setStateCallBack) first should open, second should close. 1`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing toggle(setStateCallBack) first should open, second should close. 2`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-open uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
  <forwardRef(Popover)
    controlled={false}
    location="local"
  />
</div>
`;

exports[`UU5.Bricks.Dropdown interface testing toggle(setStateCallBack) first should open, second should close. 3`] = `
<div
  className="color-schema-blue uu5-bricks-dropdown uu5-bricks-dropdown-dropdown uu5-bricks-dropdown-dropdown"
  id="uuID01"
  onMouseLeave={null}
  onMouseOver={null}
  style={Object {}}
  title={null}
>
  <div
    className="uu5-bricks-dropdown-button-cover"
    id="uuID01-cover"
  >
    <Button
      className="uu5-bricks-dropdown-button"
      colorSchema="blue"
      content={
        Array [
          "Dropdown",
          <Icon
            authenticate={false}
            className={null}
            clickable={true}
            controlled={true}
            disabled={false}
            hidden={false}
            icon="mdi-menu-down"
            id={null}
            mainAttrs={null}
            name={null}
            nestingLevel={null}
            noIndex={false}
            parent={null}
            pureRender={false}
            ref_={null}
            selected={false}
            src=""
            style={null}
            tooltip={null}
          />,
        ]
      }
      onClick={[Function]}
      size="l"
    />
  </div>
</div>
`;
